boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx

```diff
@@ -158,24 +144,13 @@ export const DemoShowroomScreen: FC<DemoTabScreenProps<"DemoShowroom">> =
     const $drawerInsets = useSafeAreaInsetsStyle(["top"])
 
     return (
-      <DrawerLayout
-        ref={drawerRef}
-        drawerWidth={Platform.select({ default: 326, web: Dimensions.get("window").width * 0.3 })}
+      <Drawer
+        open={open}
+        onOpen={() => setOpen(true)}
+        onClose={() => setOpen(false)}
         drawerType={"slide"}
         drawerPosition={isRTL ? "right" : "left"}
-        overlayColor={open ? colors.palette.overlay20 : "transparent"}
-        onDrawerSlide={(drawerProgress) => {
-          progress.value = open ? 1 - drawerProgress : drawerProgress
-        }}
-        onDrawerStateChanged={(newState: DrawerState, drawerWillShow: boolean) => {
```

Comment:
Yeah, it's all handled by the `useDrawerProgress()` hook. Pretty neat, but it has to be called from a component thats inside the drawer so that's why the progress animation handler moved from the screen to the hamburger icon component